### PR TITLE
Antag HUDs will now animate if you have multiple antagonists, and will no longer clear on deconversion

### LINKED ISSUE #59767: Antag HUDs appear to be overwriting the old ones, and never come back.
## Reproduction:
1 - Be an antag with a unique HUD (Wizard, Cultist, Revolutionary, Nuclear operative, Fugitive/Space cop)
2 - Be converted to another antag (Cult/Revs)
3 - Get deconverted
4 - As your old hud appears to be overwritten entirely, not being saved anywhere, and your new one is deleted now that you've been deconverted, you're an antag without any antag hud.

image
image

It also fucks over Admins from being able to tell someone's an antag with Combo HUD, which could lead to accidentally aprisoning hijackers because they didn't know the Combo HUD doesn't properly show antags.
image

### PR DESCRIPTION
**Pull request by Mothblocks**
## About The Pull Request

This is for the admin combo HUD. Players shouldn't notice any difference (except at roundend).

https://user-images.githubusercontent.com/35135081/141036990-e32af38c-192f-40fd-8462-dcb27736cbb8.mp4

Removes the ability to set custom antag HUDs for custom admin teams for complexity, though if there's a large enough demand I can try to bring it back in another PR.

Fixes #59767

TM candidate only so that I can make sure antags aren't getting leaked to people who shouldn't see them.

## Changelog

:cl:
fix: Antag HUDs will no longer clear on deconversion if the player was another antag.
qol: Antag HUDs (as seen by admins and at the round end) will now animate between all antagonists, rather than just choosing the most latest.
/:cl:

### CODE REVIEW

**Comment by optimumtact** [comment]
looks too good

**Comment by tralezab** [comment]
someones gotta pull the damn keyboard away from this madman

File: code/modules/antagonists/_common/antag_hud.dm [hunk 1]
```diff
@@ -1,58 +1,78 @@
...
+			continue
+		final_antag_hud_images += image('icons/mob/hud.dmi', mind.current, antagonist.antag_hud_name)
+
+	return final_antag_hud_images
+
+/datum/atom_hud/alternate_appearance/basic/antagonist_hud/proc/update_icon()
+	if (antag_hud_images.len == 0)
+		image.icon = icon('icons/blanks/32x32.dmi', "nothing")
+	else
+		image.icon = icon('icons/mob/hud.dmi', get_antag_image(index).icon_state)
```
Comment (JohnFulpWillard) [inline code/modules/antagonists/_common/antag_hud.dm:70]: Just as a small suggestion, can the dmi file be de-hardcoded?

Comment (tralezab) [reply code/modules/antagonists/_common/antag_hud.dm:70]: i kinda like the idea of making antag huds their own dmi instead

**Comment by ThePyroTechnician** [comment]
oranges said to thumbs down this so I thumbs upped

**Comment by Med-Ike** [comment]
Better code than anything oranges has written

**Comment by Iamgoofball** [comment]
Does this apply to players? If I'm a cultist and also a Tojo Clan recruit, does it work for me as a player?

**Comment by Mothblocks** [comment]
No, that should remain normal. It's fairly tricky to support that, but something I want to do at some point.

**Comment by Kylerace** [comment]
@MrDoomBringer so many downvotes

**Comment by Rohesie** [comment]
We cannot.
In an effort to better cater to the Reddit users we are now closing and denying any PR with more downvotes than upvotes.

**Comment by Mothblocks** [comment]
Me and Rohesie are now mortal enemies and I will now fragrantly abuse my power to reopen this PR in a desperate bid of showmanship

**Comment by KIBORG04** [comment]
11 upvotes vs 10 downvotes. merge now

**Comment by Mothblocks** [comment]
It works

**Pull request merged**